src/index.ts

```diff
@@ -205,13 +206,41 @@ class ForkTsCheckerWebpackPlugin {
 
     this.validateVersions();
 
-    this.vue = options.vue === true; // default false
+    if (options.vue && options.externalProgramHandler) {
+      throw new Error(
+        'When you use the `vue` option, you cannot use the `externalProgramHandler` option!'
+      );
+    } else if (options.vue) {
+      this.pluggableProgramFactory = __dirname + '/VueProgramFactory';
+    } else if (options.externalProgramHandler) {
+      this.pluggableProgramFactory = require.resolve(
+        options.externalProgramHandler,
+        {
+          paths: [
+            ...module.paths,
+            // also try to resolve relative to parent module (in case fork-ts-checker-webpack-plugin was symlinked, e.g. via `npm link`)
+            ...(module.parent ? module.parent.paths : [])
+          ]
+        }
+      );
+    }
 
     this.useTypescriptIncrementalApi =
       options.useTypescriptIncrementalApi === undefined
-        ? semver.gte(this.typescriptVersion, '3.0.0') && !this.vue
+        ? semver.gte(this.typescriptVersion, '3.0.0') &&
+          !this.pluggableProgramFactory
         : options.useTypescriptIncrementalApi;
 
+    /*
```

Comment:
As the comment says - I would add a warning like this, but it would break current behaviour.